aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -298,60 +298,63 @@ void bernoulli_scalar_kernel(Tensor& self, double p, c10::optional<Generator> ge
 }
 #else
 void bernoulli_scalar_kernel(Tensor &self, double p, c10::optional<Generator> gen) {
-  if (cpuinfo_initialize() && cpuinfo_vendor_intel == cpuinfo_get_processor(0)->core->vendor) {
-    CPUGeneratorImpl* generator = get_generator_or_default<CPUGeneratorImpl>(gen, detail::getDefaultCPUGenerator());
-    int64_t seed;
-    {
-      // See Note [Acquire lock when using random generators]
-      std::lock_guard<std::mutex> lock(generator->mutex_);
-      seed = generator->random();
+  // FIXME: Is a check really needed here? Should fail to compile if no MKL
+  if (!cpuinfo_initialize()) { AT_ASSERTM(false, "cpuinfo failed to initialize"); }
+  const auto vendor = cpuinfo_get_processor(0)->core->vendor;
+  if (vendor != cpuinfo_vendor_intel && vendor != cpuinfo_vendor_amd) {
```

Comment:
This doesn't really belong  in this PR but I can't test bernoulli without it. I can't see why we should disallow compiling with MKL on AMD CPUs.